mobile/lib/features/trade/trade_screen.dart

```diff
@@ -256,3 +275,26 @@ class TradeScreen extends StatelessWidget {
         ));
   }
 }
+
+class LatestPriceWidget extends StatelessWidget {
+  final Usd price;
+  final String label;
+
+  const LatestPriceWidget({super.key, required this.price, required this.label});
+
+  @override
+  Widget build(BuildContext context) {
+    return RichText(
+      text: TextSpan(
+        text: label,
+        style: DefaultTextStyle.of(context).style,
+        children: [
+          TextSpan(
+            text: "\$${price.formatted()}",
```

Comment:
🔧 i think the to string implement action is doing that already so you could simply 

```suggestion
            text: "$price",
```